Sources/OpenCombine/Publishers/Publishers.DropWhile.swift

```diff
@@ -46,88 +44,132 @@ extension Publishers {
         public func receive<SubscriberType: Subscriber>(subscriber: SubscriberType)
             where Output == SubscriberType.Input, SubscriberType.Failure == Error
         {
-            let dropWhile = _DropWhile<Upstream, SubscriberType, (Output) throws -> Bool>(
-                downstream: subscriber, predicate: predicate
-            )
-
-            upstream.receive(subscriber: dropWhile)
+            let inner = Inner(downstream: subscriber, predicate: catching(predicate))
+            upstream.receive(subscriber: inner)
         }
     }
 }
 
-private final class _DropWhile<Upstream: Publisher, Downstream: Subscriber, Predicate>
-    : Subscriber,
+private class _DropWhile<Upstream: Publisher, Downstream: Subscriber>
```

Comment:
Hey, great work :)
Could you please make these changes in a separate PR? These changes don't have a relation to `tryMap` operator

Reply:
This PR is now about `Inner` refactoring, for `TryMap` see #15.

#15 depends on this PR, so it's better to merge this first, and review #15 after that.